tools/chapel-py/src/chapel/replace/__init__.py

```diff
@@ -121,10 +122,28 @@ def rename_named_actuals(rc, call, renames):
             # yield (actual, actual_text.replace(name, renames[name]))
             yield from []
 
-def _do_replace(finder, ctx, filename, suffix, inplace):
+def _init_content(rc: ReplacementContext) -> str:
+    return rc.content
+
+def _update_content(content: str, replace_from: int, replace_to: int, replace_with: str) -> str:
+    return content[:replace_from] + replace_with + content[replace_to:]
+
+
+def replace(finder: typing.Generator,
+            ctx: chapel.core.Context,
+            filename: str,
+            init_content: typing.Callable[[ReplacementContext], typing.Any] = _init_content,
+            update_content: typing.Callable[[typing.Any, int, int, typing.Any], typing.Any] = _update_content) -> typing.Any:
+    """
+    Drives replacement of text based on matches found in `finder`.
+    By default the content to be replaced is represented by a string, however this can be changed by providing values for `init_content` and `do_update`.
+
+    `init_content` provides the initial value for the new text based on the ReplacementContext
+    `do_update` updates the new text, given the given the current content and what and where to replace
+    """
```

Comment:
These callables seem a little bit awkward -- do you have a use case? Maybe we can come up with something more ergonomic.

Reply:
I had a usecase when I first created these, but that has since gone away. I agree they are a little awkward to use and would not be opposed to just removing them from this PR, rather than trying to design something better right now

Reply:
I think I would prefer removing them if you don't have a use case anymore (but I am curious what it is!)